### PR TITLE
Add changelog and docs for document task

### PR DESCRIPTION
**Pull request by VictorZheng-qizheng**
Add changelog and docs for items related to document task in Excel.

### CODE REVIEW

File: api-reference/beta/resources/workbookworksheet.md [hunk 1]
```diff
@@ -34,6 +34,7 @@ An Excel worksheet is a grid of cells. It can contain data, tables, charts, etc.
 |[List](../api/worksheet-list.md) | [workbookWorksheet](workbookworksheet.md) collection |Get worksheet object collection. |
 |[Add](../api/worksheetcollection-add.md)|[workbookWorksheet](workbookworksheet.md)|Adds a new worksheet to the workbook. The worksheet will be added at the end of existing worksheets. |
 |[List pivotTables](../api/workbookworksheet-list-pivottables.md) |[workbookPivotTable](workbookpivottable.md) collection| Get a workbookPivotTable object collection.|
+|[List tasks](../api/workbookworksheet-list-tasks.md)|[workbookDocumentTask](workbookdocumenttask.md) collection|Get the workbookDocumentTask object collection.|
```
Comment (AmandaHan123) [inline api-reference/beta/resources/workbookworksheet.md:37]: Also add to Relationships in this page?

File: api-reference/beta/resources/workbookdocumenttask.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+|:---|:---|:---|
+|[List changes](../api/workbookdocumenttask-list-changes.md)|[workbookDocumentTaskChange](workbookdocumenttaskchange.md) collection|Get the workbookDocumentTaskChange resources from the changes navigation property.|
+
+## Properties
+|Property|Type|Description|
+|:---|:---|:---|
+|assignees|[workbookEmailIdentity](workbookemailidentity.md) collection| A collection of user identity the task is assigned to|
+|completedBy|[workbookEmailIdentity](workbookemailidentity.md)|A user identity that completes the task|
+|completedDateTime|DateTimeOffset|Timestamp of the task completion|
+|createdBy|[workbookEmailIdentity](workbookemailidentity.md)|A useridentity that creates the task|
```
Comment (AmandaHan123) [inline api-reference/beta/resources/workbookdocumenttask.md:32]: useridentity -> user identity

File: api-reference/beta/resources/workbookdocumenttask.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+## Methods
+|Method|Return type|Description|
+|:---|:---|:---|
+|[List changes](../api/workbookdocumenttask-list-changes.md)|[workbookDocumentTaskChange](workbookdocumenttaskchange.md) collection|Get the workbookDocumentTaskChange resources from the changes navigation property.|
+
+## Properties
+|Property|Type|Description|
+|:---|:---|:---|
+|assignees|[workbookEmailIdentity](workbookemailidentity.md) collection| A collection of user identity the task is assigned to|
+|completedBy|[workbookEmailIdentity](workbookemailidentity.md)|A user identity that completes the task|
```
Comment (miwoznic) [inline api-reference/beta/resources/workbookdocumenttask.md:30]: Should you also say which of the properties on this object are nullable? Or is that not the convention in these docs?

File: api-reference/beta/resources/workbookdocumenttask.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+  "keyProperty": "id",
+  "@odata.type": "microsoft.graph.workbookDocumentTask",
+  "baseType": "microsoft.graph.entity",
+  "openType": false
+}
+-->
+``` json
+{
+  "@odata.type": "#microsoft.graph.workbookDocumentTask",
+  "id": "String (identifier)",
```
Comment (miwoznic) [inline api-reference/beta/resources/workbookdocumenttask.md:59]: In some files you have actual sample values, and in others you just list the property type like you did here. Is there an established convention for what the json representation ought to look like for the graph docs, or does it not matter?

File: api-reference/beta/resources/workbookdocumenttaskchange.md [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+Inherits from [entity](../resources/entity.md).
+
+## Methods
+
+
+
+## Properties
+|Property|Type|Description|
+|:---|:---|:---|
+|assignee|[workbookEmailIdentity](workbookemailidentity.md)|Only exists on assign change history|
```
Comment (miwoznic) [inline api-reference/beta/resources/workbookdocumenttaskchange.md:28]: I think the descriptions here should be more clear about when the properties are null or not. Maybe rephrase the description to something like "Only present when the `type` property is `assign`".

File: api-reference/beta/resources/workbookdocumenttaskchange.md [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+|changedBy|[workbookEmailIdentity](workbookemailidentity.md)|The user who performs the change|
+|commentId|String|The identifier of the associated comment|
+|createdDateTime|DateTimeOffset|A timestamp of the change|
+|dueDateTime|DateTimeOffset|Only exists on setSchedule change history |
+|id|String|The identifier of the change history|
+|percentComplete|Int32|Only exists on setPercentComplete change history|
+|priority|Int32|Only exists on setPriority change history|
+|startDateTime|DateTimeOffset|Only exists on setSchedule change history|
+|title|String|Only exists on setTitle change history|
+|type|String|Type of the change history, including create, assign, unassign, unassignAll, setPriority, setTitle, setPercentComplete, setSchedule, remove, restore, undo, and unknownFutureValue |
```
Comment (miwoznic) [inline api-reference/beta/resources/workbookdocumenttaskchange.md:38]: I would phrase this as something like "Type of the change history. Possible values include `create`, `assign`, etc."

File: api-reference/beta/resources/workbookemailidentity.md [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Represents an identity type
+
+## Properties
+|Property|Type|Description|
+|:---|:---|:---|
+|displayName|String| Display name of the user|
+|email|String|Email address of the user|
+|id|String|Identifier of the user identity|
```
Comment (miwoznic) [inline api-reference/beta/resources/workbookemailidentity.md:23]: Nit: I would phrase this as "Unique identifier of the user"

File: api-reference/beta/resources/workbookworksheet.md (hunk 1 above)
Comment (miwoznic) [inline api-reference/beta/resources/workbookworksheet.md:37]: I would be specific and say "Get the workbookDocumentTask object collection containing all document tasks on this worksheet"

**Review by miwoznic — CHANGES_REQUESTED** [review]

File: api-reference/beta/resources/workbookcommentreply.md [hunk 8]
```diff
@@ -31,7 +31,9 @@ Represents a reply to an excel comment.
 
 ## Relationships
 
-None
+|Relationship|Type|Description|
+|:---|:---|:---|
+|task|[workbookDocumentTask](workbookdocumenttask.md)|Task associated with the comment thread|
```
[the remark this answers is not in the document]
Comment (VictorZheng-qizheng) [reply api-reference/beta/resources/workbookcommentreply.md:36]: Tasks are always associated with the threads. The task associated with the top-level comment is returned.

File: api-reference/beta/api/workbookworksheet-list-tasks.md [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+
+Retrieve a list of workbookdocumenttask objects.
+
+## Permissions
+One of the following permissions is required to call this API. To learn more, including how to choose permissions, see [Permissions](/graph/permissions-reference).
+
+
+|Permission type      | Permissions (from least to most privileged)              |
+|:--------------------|:---------------------------------------------------------|
+|Delegated (work or school account) | Files.ReadWrite    |
```
[the remark this answers is not in the document]
Comment (VictorZheng-qizheng) [reply api-reference/beta/api/workbookworksheet-list-tasks.md:24]: ReadWrite is required to get comments and worksheets, so I assume it's also required to get the tasks.

File: api-reference/beta/api/workbookdocumenttask-list-changes.md [hunk 10]
```diff
@@ -0,0 +1,116 @@
...
+
+Retrieve a list of workbookdocumenttaskchange objects.
+
+## Permissions
+One of the following permissions is required to call this API. To learn more, including how to choose permissions, see [Permissions](/graph/permissions-reference).
+
+
+|Permission type      | Permissions (from least to most privileged)              |
+|:--------------------|:---------------------------------------------------------|
+|Delegated (work or school account) | Files.ReadWrite    |
```
[the remark this answers is not in the document]
Comment (VictorZheng-qizheng) [reply api-reference/beta/api/workbookdocumenttask-list-changes.md:24]: ReadWrite is required to get comments and worksheets, so I assume it's also required to get the tasks.

[the remark this answers is not in the document]
Comment (VictorZheng-qizheng) [reply api-reference/beta/api/workbookdocumenttask-list-changes.md:24]: I am changing MSA to not supported.

File: api-reference/beta/resources/workbookdocumenttask.md (hunk 4 above)
Comment (VictorZheng-qizheng) [reply api-reference/beta/resources/workbookdocumenttask.md:59]: The descriptions of the entities have the json representation, while the api descriptions for the methods have the example response.

File: api-reference/beta/api/workbookdocumenttask-list-changes.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (miwoznic) [reply api-reference/beta/api/workbookdocumenttask-list-changes.md:24]: I know for a fact that some of the graph API permissions are documented incorrectly, e.g. the [Get Range](https://learn.microsoft.com/en-us/graph/api/range-get?view=graph-rest-1.0&tabs=http) API says Files.ReadWrite is required but I tested it in the past in Graph explorer and only Files.Read was required. Do you know how these permissions are determined? Is there some config from the Graph side that controls access based on permissions, or is it up to the workload (i.e. RichAPI) to control access?

File: api-reference/beta/resources/workbookdocumenttaskchange.md [hunk 11]
```diff
@@ -25,18 +25,18 @@ Inherits from [entity](../resources/entity.md).
...
-|priority|Int32|Only exists on setPriority change history|
-|startDateTime|DateTimeOffset|Only exists on setSchedule change history|
-|title|String|Only exists on setTitle change history|
-|type|String|Type of the change history, including create, assign, unassign, unassignAll, setPriority, setTitle, setPercentComplete, setSchedule, remove, restore, undo, and unknownFutureValue |
-|undoChangeId|String|Only exists on undo change history |
+|percentComplete|Int32|Only present when the `type` property is `setPercentComplete`. Nullable.|
+|priority|Int32|Only present when the `type` property is `setPriority`. Nullable.|
+|startDateTime|DateTimeOffset|Only present when the `type` property is `setSchedule`. Nullable.|
+|title|String|Only present when the `type` property is `setTitle`. Nullable.|
+|type|String|Type of the change history. Possible values include create, assign, unassign, etc.|
```
Comment (miwoznic) [inline api-reference/beta/resources/workbookdocumenttaskchange.md:38]: Are all of the possible values documented somewhere? If not, I think you should list out all of the possible values here.

**Review by miwoznic — APPROVED** [review]

**Comment by JarbasHorst** [comment]
Hello @VictorZheng-qizheng, can you please share with me a pointer (the URLs) to the CSDL and api.md files related to your PR? Thanks!

**Comment by VictorZheng-qizheng** [comment]
> Hello @VictorZheng-qizheng, can you please share with me a pointer (the URLs) to the CSDL and api.md files related to your PR? Thanks!

@JarbasHorst Sure.
Link to api.md: https://microsoftgraph.visualstudio.com/onboarding/_git/onboarding?version=GBmaster&path=/reviews/4065-Excel-tasks-API-For-Sync/api.md

Link to CSDL workload PR: https://msazure.visualstudio.com/One/_git/AD-AggregatorService-Workloads/pullrequest/7993031?path=/Workloads/Microsoft.Excel/override/schema-Prod-beta.csdl&discussionId=145664075

File: changelog/Microsoft.Excel.json [hunk 12]
```diff
@@ -1,5 +1,103 @@
...
+          "ChangeType": "Addition",
+          "Description": "Added the [count](https://learn.microsoft.com/en-us/graph/api/workbookDocumentTask?view=graph-rest-beta) method to the [workbookDocumentTask](https://learn.microsoft.com/en-us/graph/api/workbookDocumentTask?view=graph-rest-beta) resource.",
+          "Target": "workbookDocumentTask"
+        },
+        {
+          "Id": "5472d980-5acf-4663-b1af-c7d22e28d1b3",
+          "ApiChange": "Method",
+          "ChangedApiName": "count",
+          "ChangeType": "Addition",
+          "Description": "Added the [count](https://learn.microsoft.com/en-us/graph/api/workbookDocumentTaskChange?view=graph-rest-beta) method to the [workbookDocumentTaskChange](https://learn.microsoft.com/en-us/graph/api/workbookDocumentTaskChange?view=graph-rest-beta) resource.",
```
Comment (JarbasHorst) [inline changelog/Microsoft.Excel.json:74]: I don't see any associated documentation update for this and the following methods. Please, update the docs accordingly.

File: api-reference/beta/resources/workbookdocumenttaskchange.md [hunk 13]
```diff
@@ -0,0 +1,75 @@
...
+Represents a task change history of a [workbookDocumentTask](workbookdocumenttask.md)
+
+
+Inherits from [entity](../resources/entity.md).
+
+## Methods
+
+
+
+## Properties
```
Comment (JarbasHorst) [inline api-reference/beta/resources/workbookdocumenttaskchange.md:25]: The api.md file mentions the documentTaskChangeAction enum that is not documented in this PR. Apparently, this enum has the same values as the type property. Can you please help clarity whether the documentTaskChangeAction enum should be documented or not? Thanks

**Comment by JarbasHorst** [comment]
Hello @VictorZheng-qizheng, I reviewed your PR and left some comments. Please, take a look when you will have time.

File: api-reference/beta/resources/workbookdocumenttaskchange.md (hunk 13 above)
Comment (VictorZheng-qizheng) [reply api-reference/beta/resources/workbookdocumenttaskchange.md:25]: documentTaskChangeAction should not be documented. `type` on the change is perceived as a string instead of this enum.

File: api-reference/beta/resources/workbookdocumenttaskchange.md [hunk 14]
```diff
@@ -0,0 +1,75 @@
+---
```
[the remark this answers is not in the document]
Comment (VictorZheng-qizheng) [reply api-reference/beta/resources/workbookdocumenttaskchange.md:1]: GET will be included in this next iteration.
POST is hidden and should not be available to third party users as mentioned on api.md.  
LIST DocumentTaskChange is documented on workbookdocumenttask-list-changes.md

File: changelog/Microsoft.Excel.json (hunk 12 above)
Comment (VictorZheng-qizheng) [reply changelog/Microsoft.Excel.json:74]: How do I document count and ItemAt for a collection? I don't find these two methods in the documentation. I am going to the remove them from the changelog
